Issues/168 (#303)
V4Fire/Client: src/core/dom/image/README.md

# core/dom/image

This module provides API to load images by using `background-image` or `src`.

## Callbacks

| Name       | Description                                      | Payload description  | Payload   |
| ---------- |------------------------------------------------- | -------------------- |-----------|
| `load`     | Invoked when an image was successfully loaded    | `el` bonded node     | `Element` |

> | `error`    | Invoked when a loading error of an image appears | `el` bonded node     | `Element` |

Reviewer: Здесь нет ошибки? `error` `el` bonded node     | `Element`

## Usage

### Basic

```typescript
import { ImageLoader } from 'core/dom/image';

@component()
export default class bSomeComponent extends iBlock {
  @hook('mounted')
  initImage(): void {
    ImageLoader.init(<HTMLDivElement | HTMLImageElement>this.$el, {
      src: 'https://img.src'
    })
  }
}
```

### Using callbacks

```typescript
import { ImageLoader } from 'core/dom/image';

@component()
export default class bSomeComponent extends iBlock {
  @hook('mounted')
  initImage(): void {
    ImageLoader.init(this.$el, {
      src: 'https://img.src',
      ctx: this,
      load: (el) => this.doSomething(el),
      error: (el) => alert('help')
    })
  }
}
```

> It is highly recommended to specify the context if you are using callbacks explicitly.

### Additional stages

Stages of an image:

> 1. `preview` – It will be displayed until the main image is loading.

Reviewer: ```suggestion
1. `preview` – the main image is loading; till the loading complete, there will be shown a placeholder
```

> 2. `main` – It will be displayed until then the main image is loaded.

Reviewer: ```suggestion
2. `main` – the main image has been loaded.
```

> 3. `broken` – It will be displayed if the main image failed to load.

Reviewer: ```suggestion
3. `broken` – the main image hasn't been loaded due to an error; there will be shown an error placeholder.
```

```typescript
import { ImageLoader } from 'core/dom/image';

@component()
export default class bSomeComponent extends iBlock {
  @hook('mounted')
  initImage(): void {
    ImageLoader.init(this.$el, {
      src: 'https://img.src',
      preview: 'https://preview.src',
      broken: 'https://broken.src'
    })
  }
}
```

### Different image formats

> You can use several different image formats or resolutions by using the `srcset` and `sources` options.

Reviewer: ```suggestion
You can use several image formats or resolutions by using the `srcset` and `sources` options.
```

```typescript
import { ImageLoader } from 'core/dom/image';

@component()
export default class bSomeComponent extends iBlock {
  @hook('mounted')
  initImage(): void {
    ImageLoader.init(this.$el, {
      src: 'https://img.src',
      sources: [{
        type: 'webp',
        sources: [{srcset: 'https://img-webp.src', type: 'webp'}]
      }]
    })
  }
}
```

### Default value for stage images

> To not install a preview or a broken image every time, it is possible to set theDefault settings.

Reviewer: To avoid redundant lines of code, you can specify default parameters for load and error stages of an image.

> Override the `const.ts` file in your layer in the `core/dom/image` folder.

Reviewer: ```suggestion
To do this, you have to override the `const.ts` file within the `core/dom/image` folder in your project.
```

> __const.ts__

Reviewer: ```suggestion
**const.ts**

```

# core/dom/image

> This module provides API to set a `background-image` or an `src` to nodes.

Reviewer: ```suggestion
This module provides API to load images by using `background-image` or `src`.
```

## Callbacks

| Name       | Description                                  | Payload description  | Payload   |
| ---------- |----------------------------------------------| -------------------- |-----------|
| `load`     | Invoked after successful loading of an image | `el` bonded node     | `Element` |

> | `error`    | Invoked if loading error appears             | `el` bonded node     | `Element` |

Reviewer: Вторая строчка содержит ошибки в описании

> ## Usage

Reviewer: А где примеры использование brokenImage и иконки загрузки?

```typescript
import { ImageLoader } from 'core/dom/image';

@component()
export default class bSomeComponent extends iBlock {
  @hook('mounted')
  initImage(): void {
    ImageLoader.init(<HTMLDivElement | HTMLImageElement>this.$el, {

> 		src: 'https://img.src'

Reviewer: Кажется беда с отступами (в md мы юзаем пробелы)
